gradio/routes.py

```diff
@@ -1218,32 +1241,21 @@ async def analytics_dashboard(key: str):
 ########
 
 
-def safe_join(directory: str, path: str) -> str:
-    """Safely path to a base directory to avoid escaping the base directory.
-    Borrowed from: werkzeug.security.safe_join"""
-    _os_alt_seps: List[str] = [
-        sep for sep in [os.path.sep, os.path.altsep] if sep is not None and sep != "/"
-    ]
-
+def routes_safe_join(directory: DeveloperPath, path: UserProvidedPath) -> str:
+    """Safely join the user path to the directory while performing some additional http-related checks."""
```

Comment:
Nice this makes sense. Perhaps worth mentioning the checks explicitly here:
```suggestion
    """Safely join the user path to the directory while performing some additional http-related checks, e.g. ensuring that the full path exists on the local file system and is not a directory."""
```